website/docs/guides/testing.md

```diff
@@ -3,70 +3,486 @@ id: testing
+
+  <TabItem value="jest">
+
+  ```
+  npm install --save-dev @effection/jest
+  ```
+
+  </TabItem>
+  <TabItem value="mocha">
+
+  ```
+  npm install --save-dev @effection/mocha
+  ```
+
+  </TabItem>
+</Tabs>
+
+### Writing Tests
+
+Let's say we have a test case written without the benefit of effection that
+looks like this:
+
+``` javascript
+describe("a server", () => {
+  let server: Server;
+  beforeEach(async () => {
+    server = await startServer({ port: 3500 });
+  });
+
+  it('can be pinged', async () => {
+    let response = await fetch('http://localhost:3500/ping');
+    expect(response.ok).toBe(true);
+  });
+
+  afterEach(async () => {
+    await server.close();
+  });
+});
 ```
-npm install --save-dev @effection/mocha
+:::note Cleanup
+
+It is critical that we shutdown the server after each test. Otherwise, node
+will never exit without a hard stop because it's still bound to port
+`3500`, and worse, every subsequent time we try and run our tests they will fail
+because port `3500` is not available!
+:::
+
+To re-write this test case using effection, there are two superficial
+differences from the vanilla version of the framework.
+
+1. You `import` all of your test syntax from the effection package, and not
+from the global namespace.
+2. You use [generator functions instead of async functions][replace-async] to
+express all of your test operations.
+
+Our test case now looks like this.
+
+<Tabs
+  groupId="jest-and-mocha-replacement"
+  defaultValue="jest"
+  values={[{label: 'Jest', value: 'jest'}, {label: 'Mocha', value: 'mocha'}]}>
+
+  <TabItem value="jest">
+
+``` javascript
+import { beforeEach, afterEach, it } from '@effection/jest';
+
+describe("a server", () => {
+  let server: Server;
+  beforeEach(function*() {
+    server = yield startServer({ port: 3500 });
+  });
+
+  it('can be pinged', function*() {
+    let response = yield fetch('http://localhost:3500/ping');
+    expect(response.ok).toBe(true);
+  });
+
+  afterEach(function*() {
+    yield server.close();
+  });
+});
+```
+
+  </TabItem>
+  <TabItem value="mocha">
+
+``` javascript
+import { beforeEach, afterEach, it } from '@effection/mocha';
+
+describe("a server", () => {
+  let server: Server;
+  beforeEach(function*() {
+    server = yield startServer({ port: 3500 });
+  });
+
+  it('can be pinged', function*() {
+    let response = yield fetch('http://localhost:3500/ping');
+    expect(response.ok).toBe(true);
+  });
+
+  afterEach(function*() {
+    yield server.close();
+  });
+});
+```
+
+  </TabItem>
+</Tabs>
+
+But so far, we've only traded one syntax for another. Now however, we can
+begin to leverage the power of Effection to make our test cases not only more
+concise, but also more flexible. To do this, we use the fact that each test
+case gets its own scope that is automatically destroyed for us after it runs.
+
+So if we re-cast our server as a [resource][] that is running in our test scope,
```

Comment:
no link in `resource[]`

Reply:
It's at the bottom of the document. Turns out Markdown uses the text as the link if one is not provided.